lib/shopify_transporter/shopify/customer.rb

```diff
@@ -57,12 +57,30 @@ def to_csv
       ).freeze
 
       def top_level_row_values
-        base_hash.merge(record_hash.slice(*TOP_LEVEL_ATTRIBUTES)).values
+        default_address.merge(top_level_attributes).values_at(*Customer.columns)
+      end
+
+      def top_level_attributes
+        base_hash
+          .merge(record_hash.slice(*TOP_LEVEL_ATTRIBUTES))
+          .delete_if { |_, v| v.nil? }
+      end
+
+      def default_address
+        if at_least_one_address?
```

Comment:
I'd write this slightly differently, something like:

```ruby
def default_address
  return {} unless at_lease_one_address?

  record_hash['addresses'][0].slice(*ADDRESS_ATTRIBUTES)
end

Reply:
👍 to this. We use this pattern a lot as a guard clause.